iot_verifier/src/main.rs

```diff
@@ -87,8 +87,10 @@ impl Server {
 
         let iot_config_client = IotConfigClient::from_settings(&settings.iot_config_client)?;
 
-        let gateway_cache = GatewayCache::from_settings(iot_config_client.clone());
-        _ = gateway_cache.prewarm().await;
+        let gateway_updater =
```

Comment:
the gateway_updater struct really doesn't need a copy of the `MessageReceiver` since it has no use for it; i think it's more idiomatic to have the `GatewayUpdater::from_settings()` return a tuple `(gateway_update_receiver, gateway_updater) = GatewayUpdater::from_settings(...inputs...)` and then only pass the receiver clones to the things that need them.

Reply:
I actually like `from_settings` returning a tuple, would it be better if it was a free function so you called it via `gateway_update::from_settings` rather than being part of the `GatewayUpdater` impl?

Reply:
I dont have a better suggestion so addressed here: 4b05c721c6c7d8f403ebf4ffeb183386973c9d2a